packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+import * as Sentry from "@sentry/serverless";
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { ResourceTypeForConsolidation } from "@metriport/api-sdk";
+import { makeLambdaClient } from "@metriport/core/external/aws/lambda";
+import { makeS3Client } from "@metriport/core/external/aws/s3";
+import { DOMParser } from "xmldom";
+import { capture } from "./shared/capture";
+import { getEnvOrFail } from "./shared/env";
+import { prefixedLog } from "./shared/log";
+import { postToConverter } from "./shared/converter";
+
+// Keep this as early on the file as possible
+capture.init();
+
+// Automatically set by AWS
+const lambdaName = getEnvOrFail("AWS_LAMBDA_FUNCTION_NAME");
+const region = getEnvOrFail("AWS_REGION");
+// Set by us
+const axiosTimeoutSeconds = Number(getEnvOrFail("AXIOS_TIMEOUT_SECONDS"));
+const bucketName = getEnvOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+// converter config
+const FHIRToCDAConverterUrl = getEnvOrFail("FHIR_TO_CDA_CONVERTER_URL");
+const convertDocLambda = getEnvOrFail("CONVERT_DOC_LAMBDA_NAME");
+const converterKeysTableName = getEnvOrFail("FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME");
+
+const lambdaClient = makeLambdaClient(region);
+const s3Client = makeS3Client(region);
+
+export const handler = Sentry.AWSLambda.wrapHandler(
+  async (req: {
+    bundle: Bundle<Resource>;
+    patientId: string;
+    cxId: string;
+    resources?: ResourceTypeForConsolidation[];
+    dateFrom?: string;
+    dateTo?: string;
+    conversionType: string;
+  }) => {
+    const { bundle, patientId, cxId, resources, dateFrom, dateTo, conversionType } = req;
+
+    try {
+      const log = prefixedLog(`patient ${patientId}`);
+      const res = await postToConverter({
+        url: FHIRToCDAConverterUrl,
+        payload: bundle,
+        converterKeysTableName,
+        axiosTimeoutSeconds,
+        log,
+      });
+
+      const formattedXML = formatXML(res.data);
+      const fileName = createFileName({ cxId, patientId, resources, dateFrom, dateTo });
+
+      await s3Client
+        .putObject({
+          Bucket: bucketName,
+          Key: fileName,
+          Body: formattedXML,
+          ContentType: "application/xml",
+        })
+        .promise();
+
+      const convertUrl = await convertDoc({ fileName, conversionType });
```

Comment:
Isn't this to convert from XML to PDF? If so, we would only call this when `conversionType` is `pdf`, no?